openapi-codegen/core/src/main/scala/sttp/tapir/codegen/EndpointGenerator.scala

```diff
@@ -19,49 +19,60 @@ class EndpointGenerator {
 
   private[codegen] def allEndpoints: String = "generatedEndpoints"
 
-  def endpointDefs(doc: OpenapiDocument): String = {
+  def endpointDefs(doc: OpenapiDocument, useHeadTagForObjectNames: Boolean): Map[Option[String], String] = {
     val components = Option(doc.components).flatten
-    val ge = doc.paths.flatMap(generatedEndpoints(components))
-    val definitions = ge
-      .map { case (name, definition) =>
-        s"""|lazy val $name =
+    val geMap =
+      doc.paths.flatMap(generatedEndpoints(components, useHeadTagForObjectNames)).groupBy(_._1).mapValues(_.map(_._2).reduce(_ ++ _))
```

Comment:
This would've been so much cleaner with `groupMapReduce(_._1)(_._2)(_ ++ _)` 😂 but, alas, scala 2.12...